src/Resources/public/gtm.enhancedEcommerce.checkout.js

```diff
@@ -1,44 +1,56 @@
-(function ( $ ) {
-    'use strict';
+(function ($) {
+  'use strict';
 
-    var CHECKOUT_STEP_SHIPPING = 3;
-    var CHECKOUT_STEP_PAYMENT = 4;
-
-    $.fn.extend({
-        enhancedEcommerceCheckoutShipping: function () {
-            $(this).on('submit', function () {
-                enhancedEcommerceTrackCheckoutOption(CHECKOUT_STEP_SHIPPING, $('input[type=radio]:checked', $(this)).val());
-            });
-        },
-
-        enhancedEcommerceCheckoutPayment: function () {
-            $(this).on('submit', function () {
-                enhancedEcommerceTrackCheckoutOption(CHECKOUT_STEP_PAYMENT, $('input[type=radio]:checked', $(this)).val());
-            });
-        }
-    });
+  if (typeof checkoutStepsConfiguration !== "undefined" && checkoutStepsConfiguration.enabled === true) {
```

Comment:
I would prefer some early returns here to improve readability. Maybe we can switch the if statements to return early in case we don't accept the criteria. Probably makes this JS piece easier to read, now you have to read if, then another if, then another if. A flow like this reads easier:

```javascript
if (!bla) {
     return;
}

if (!blaSecond) {
    return;
}

.. perform action here
```